### PR TITLE
[Chore] Fix linter issue for golangci-lint 1.51.2

### PR DESCRIPTION
**Pull request by ying-jeanne**
#### Which issue(s) this PR fixes or relates to

Fixes #<issue number>

#### Checklist

- [x] Tests updated
- [x] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -44,6 +44,7 @@ Querying with using `{__mimir_storage__="ephemeral"}` selector no longer works.
...
 * [ENHANCEMENT] Memcached: added `-*.memcached.min-idle-connections-headroom-percentage` support to configure the minimum number of idle connections to keep open as a percentage (0-100) of the number of recently used idle connections. This feature is disabled when set to a negative value (default), which means idle connections are kept open indefinitely. #4249
+* [BUGFIX] Linter: fix tests that fails on golangci-lint v1.51.2. #4278
```
Comment (aknuds1) [inline CHANGELOG.md:47]: I personally think this should not be called out as a bug fix, since it doesn't affect users.

**Review by aknuds1 — COMMENTED** [review]
I don't think a changelog entry is warranted.

File: CHANGELOG.md (hunk 1 above)
Comment (ying-jeanne) [reply CHANGELOG.md:47]: I was not sure about it, let me remove it then.

**Review by aknuds1 — APPROVED** [review]

File: pkg/compactor/compactor_test.go [hunk 2]
```diff
@@ -1900,6 +1900,7 @@ func TestOwnUser(t *testing.T) {
 	}
 
 	for name, tc := range testCases {
+		tc := tc
```
Comment (replay) [inline pkg/compactor/compactor_test.go:1903]: if the tests called in `t.Run` run in parallel, wouldn't this lead to issues if the value in `tc` gets overwritten by the next iteration of the loop on `testCases`?

**Review by replay — APPROVED** [review]
nice, the lack of this could have hidden some failing test cases

**Comment by pstibrany** [comment]
Thank you. Hopefully Go will start doing this automatically soon.

PR descriptions would be nice too.

**Review by pstibrany — APPROVED** [review]

**Pull request merged**